api/app/utils/geospatial.py

```diff
@@ -0,0 +1,87 @@
+from dataclasses import dataclass
+import logging
+from typing import Any, Optional
+from osgeo import gdal
+
+
+logger = logging.getLogger(__name__)
+
+
+def warp_to_match_extent(source_raster: gdal.Dataset, raster_to_match: gdal.Dataset, output_path: str) -> gdal.Dataset:
+    """
+    Warp the source_raster to match the extent and projection of the other raster.
+
+    :param source_raster: the raster to warp
+    :param raster_to_match: the reference raster to match the source against
+    :param output_path: output path of the resulting raster
+    :return: warped raster dataset
+    """
+    source_geotransform = raster_to_match.GetGeoTransform()
+    x_res = source_geotransform[1]
+    y_res = -source_geotransform[5]
+    minx = source_geotransform[0]
+    maxy = source_geotransform[3]
+    maxx = minx + source_geotransform[1] * raster_to_match.RasterXSize
+    miny = maxy + source_geotransform[5] * raster_to_match.RasterYSize
+    extent = [minx, miny, maxx, maxy]
+
+    # Warp to match input option parameters
+    return gdal.Warp(output_path, source_raster, dstSRS=raster_to_match.GetProjection(), outputBounds=extent, xRes=x_res, yRes=y_res, resampleAlg=gdal.GRA_NearestNeighbour)
```

Comment:
I don't know that it matters too much at this scale, but I think for the next TPI we should reproject the DEM/TPI data to Lambert and store it that way, using something like bilinear/bicubic

Reply:
Is that because Lambert, bilinear/bicubic is less lossy when projected?

Reply:
Nothing better about Lambert, it's just what the HFI data is in and what our other DEMs are in, so it would avoid having to do any reprojecting during processing. I think just this new 50m DEM they gave us is in Albers. 
Bilinear/bicubic is best for continuous data, like elevation data.